Source/WebCore/platform/graphics/cocoa/MediaPlaybackTargetContext.mm

```diff
@@ -42,6 +42,31 @@
     m_type = MediaPlaybackTargetContext::Type::AVOutputContext;
 }
 
+MediaPlaybackTargetContext::IPCData MediaPlaybackTargetContext::ipcData() const
+{
+    if (encodingRequiresPlatformData()) {
+        if (type() == Type::AVOutputContext) {
+            if ([PAL::getAVOutputContextClass() conformsToProtocol:@protocol(NSSecureCoding)])
+                return outputContext();
+        }
+    } else
+        return NonPlatformData { deviceName(), mockState() };
+
+    return std::nullopt;
+}
+
+std::optional<MediaPlaybackTargetContext> MediaPlaybackTargetContext::fromIPCData(IPCData&& ipcData)
+{
+    if (ipcData) {
+        return WTF::switchOn(WTFMove(*ipcData), [](RetainPtr<AVOutputContext>&& outputContext) {
+            return MediaPlaybackTargetContext { outputContext.get() };
```

Comment:
should use `WTFMove(outputContext)` to avoid ref-counting churn. The constructor takes an r-value reference.